core/src/main/scala/cats/data/OptionT.scala

```diff
@@ -45,6 +68,9 @@ final case class OptionT[F[_], A](value: F[Option[A]]) {
   def semiflatMap[B](f: A => F[B])(implicit F: Monad[F]): OptionT[F, B] =
     flatMap(a => OptionT.liftF(f(a)))
 
+  def semiflatTap[B](f: A => F[B])(implicit F: Monad[F]): OptionT[F, A] =
```

Comment:
We have had this conversation before: given that the `B` has no relevance on the types of the outputs, should we just make it an existential?
```suggestion
  def semiflatTap(f: A => F[_])(implicit F: Monad[F]): OptionT[F, A] =
```

Reply:
`FlatMap` has this type parameter for `flatTap`, `EitherT` has it for it's `semiFlatTap`, so again, I would vote for consistency to give cats users a reliable experience in terms of how signatures look and when type inference works. Or do you see any other advantage of the existential which would outweigh this aspect?